mlflow/utils/checkpoint_utils.py

```diff
@@ -0,0 +1,174 @@
+        if self.save_best_only:
+            if self.monitor not in metric_dict:
+                _logger.warning(
+                    "Checkpoint logging is skipped, because checkpoint 'save_best_only' config is "
+                    "True, it requires to compare the monitored metric value, but the provided "
+                    "monitored metric value is not available."
+                )
+                return
+
+            new_monitor_value = metric_dict[self.monitor]
+            if not self._is_new_checkpoint_better(new_monitor_value):
+                # Current checkpoint is worse than last saved checkpoint,
+                # so skip checkpointing.
+                self.last_monitor_value = new_monitor_value
+                return
+
+            self.last_monitor_value = new_monitor_value
+
+        suffix = self.checkpoint_file_suffix
+        if self.save_best_only:
+            if self.save_weights_only:
+                checkpoint_model_filename = f"latest_checkpoint.weights.{suffix}"
+            else:
+                checkpoint_model_filename = f"latest_checkpoint.{suffix}"
+            checkpoint_metrics_filename = "latest_checkpoint_metrics.json"
+            checkpoint_artifact_dir = "checkpoints"
+        else:
+            if self.save_freq == "epoch":
+                sub_dir_name = f"epoch_{current_epoch}"
+            else:
+                sub_dir_name = f"global_step_{global_step}"
+
+            if self.save_weights_only:
+                checkpoint_model_filename = f"checkpoint.weights.{suffix}"
+            else:
+                checkpoint_model_filename = f"checkpoint.{suffix}"
+            checkpoint_metrics_filename = "checkpoint_metrics.json"
+            checkpoint_artifact_dir = f"checkpoints/{sub_dir_name}"
+
+        mlflow.set_tag(
+            LATEST_CHECKPOINT_ARTIFACT_TAG_KEY,
+            f"{checkpoint_artifact_dir}/{checkpoint_model_filename}",
+        )
+
+        mlflow.log_dict(
+            {**metric_dict, "epoch": current_epoch, "global_step": global_step},
+            f"{checkpoint_artifact_dir}/{checkpoint_metrics_filename}",
+        )
+
+        tmp_dir = create_tmp_dir()
+        try:
+            tmp_model_save_path = os.path.join(tmp_dir, checkpoint_model_filename)
+            self.save_checkpoint(tmp_model_save_path)
+            mlflow.log_artifact(tmp_model_save_path, checkpoint_artifact_dir)
+        finally:
+            shutil.rmtree(tmp_dir, ignore_errors=True)
+
+
+def download_checkpoint_artifact(run_id=None, epoch=None, global_step=None):
+    from mlflow.utils.mlflow_tags import LATEST_CHECKPOINT_ARTIFACT_TAG_KEY
+    from mlflow.client import MlflowClient
+
+    client = MlflowClient()
+
+    if run_id is None:
+        run = mlflow.active_run()
+        if run is None:
+            raise MlflowException(
+                "There is no active run, please provide the 'run_id' for " "'load_checkpoint' call."
```

Comment:
What does `for load_checkpoint call` mean here?

Reply:
Made it clear:
```
                "There is no active run, please provide the 'run_id' argument for "
                "'load_checkpoint' invocation."
```